spec.md

```diff
@@ -3456,11 +3470,21 @@ digit_hex = digit_dec
           | "A" | "B" | "C" | "D" | "E" | "F"
           | "a" | "b" | "c" | "d" | "e" | "f" ;
 
-(* An integer *)
-int = '"' , "b" , [ "-" ] , { digit_bin } , '"'
-    | '"' , "o" , [ "-" ] , { digit_oct } , '"'
-    | '"' , "h" , [ "-" ] , { digit_hex } , '"'
-    |             [ "-" ] , { digit_bin } ;
+(* Integer Literals *)
+int = [ "-" ] , { digit_bin } ;
+
+(* String-encoded Integer Literals *)
+int_se =
+    '"' , "b" , [ "-" ] , { digit_bin } , '"'
+  | '"' , "o" , [ "-" ] , { digit_oct } , '"'
+  | '"' , "h" , [ "-" ] , { digit_hex } , '"' ;
+
+(* Any Integer Literal *)
+int_any = int | int_se ;
+
+(* String Literals *)
+string = (* a string that is not defined and assumed to exist *) ;
+string_raw = "'" , string , "'" ;
```

Comment:
We should make `int_se` be a string in the grammar.

Reply:
I'm on the fence on this... on the one hand, I think it is super useful to be able to put in the grammar what is a legal string-encoded literal is. A reader can grok what this is without having to wonder, "What strings can I put in there?" On the other hand, this very much blurs the lines between syntax (what the grammar should be) and semantics. I tend to fall on the former side of this specific case.